coreneuron/CMakeLists.txt

```diff
@@ -129,6 +129,13 @@ if(CORENRN_ENABLE_GPU)
   # setting LANGUAGE=CXX in non-GPU builds does not work.
   list(REMOVE_ITEM CORENEURON_CODE_FILES "${CMAKE_CURRENT_SOURCE_DIR}/utils/randoms/nrnran123.cpp")
   list(APPEND CORENEURON_CODE_FILES ${CORENEURON_CUDA_FILES})
+
+  # Eigen-3.5+ provides GPU support. However, some functions cannot be called directly from within
+  # an OpenACC region. Therefore, we need to wrap them in a special API (decorate them with
+  # __device__ & acc routine tokens), which allows us to eventually call them from OpenACC.
+  if(CORENRN_ENABLE_NMODL AND EXISTS ${CORENRN_MOD2CPP_INCLUDE}/partialPivLU_api/partialPivLU.cu)
+    list(APPEND CORENEURON_CODE_FILES ${CORENRN_MOD2CPP_INCLUDE}/partialPivLU_api/partialPivLU.cu)
```

Comment:
Can I nitpick around the naming? 😃 `non` CamelCase would be better for file / directory names.
e.g. if we have `solver` directory already, it could be just `include/solver/partial_piv_lu.cu` ?

Reply:
Hi @pramodk 

The dirs structure is:
src/
&nbsp;&nbsp;&nbsp;&nbsp;-newton/
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-newton.hpp
&nbsp;&nbsp;&nbsp;&nbsp;-partialPivLU_api/
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-partialPivLU.h
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-partialPivLU.cu

You prefer like this (?):
src/
&nbsp;&nbsp;&nbsp;&nbsp;-newton/
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-newton.hpp
&nbsp;&nbsp;&nbsp;&nbsp;-partial_piv_lu_api/
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-partialPivLU.h
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;-partialPivLU.cu

Reply:
In the nmodl repository we have `src/solver/newton` and then I was thinking `src/solver/partial_piv_lu` containing `partial_piv_lu.h` and `partial_piv_lu.cu`.